src/components/ExerciseHeader.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useLayoutEffect, useState } from 'react'
```

Comment:
mostly from WriteExercise, which already had the Header and moved it here, to reuse.